bm-persona/Libraries/LibraryDetailViewController.swift

```diff
@@ -0,0 +1,362 @@
+                let newAdress = longAddress[..<ind]
+                addressLabel.text = String(newAdress)
+            } else {
+                addressLabel.text = longAddress
+            }
+        }
+        card.addSubview(addressLabel)
+        addressLabel.font = Font.thin(12)
+        addressLabel.numberOfLines = 2
+        addressLabel.textColor = UIColor.black
+        addressLabel.translatesAutoresizingMaskIntoConstraints = false
+        addressLabel.topAnchor.constraint(equalTo: nameLabel.layoutMarginsGuide.bottomAnchor, constant: 20).isActive = true
+        addressLabel.leftAnchor.constraint(equalTo: addressIcon.layoutMarginsGuide.rightAnchor, constant: 8).isActive = true
+        //addressLabel.widthAnchor.constraint(equalTo: card.layoutMarginsGuide.widthAnchor, multiplier: 0.5).isActive = true
+        addressLabel.rightAnchor.constraint(equalTo: card.layoutMarginsGuide.rightAnchor, constant: -160).isActive = true
+        addressLabel.adjustsFontSizeToFitWidth = true
+        
+        
+        let phoneLabel = UILabel()
+        phoneLabel.text = library!.phoneNumber
+        card.addSubview(phoneLabel)
+        phoneLabel.font = Font.thin(12)
+        phoneLabel.textColor = UIColor.black
+        phoneLabel.translatesAutoresizingMaskIntoConstraints = false
+        phoneLabel.topAnchor.constraint(equalTo: addressLabel.layoutMarginsGuide.bottomAnchor, constant: 20).isActive = true
+        phoneLabel.leftAnchor.constraint(equalTo: card.layoutMarginsGuide.leftAnchor).isActive = true
+        phoneLabel.widthAnchor.constraint(equalTo: card.layoutMarginsGuide.widthAnchor, multiplier: 0.5).isActive = true
+        phoneLabel.adjustsFontSizeToFitWidth = true
+        
+        //walking distance
+        if let libLat = library!.latitude, let libLong = library!.longitude {
+            let dist = calcDistance(lib_Lat: libLat, lib_Long: libLong)
+            distLabel.text = String(format: "%.1f", dist) + " miles away"
+            
+        } else {
+            distLabel.text = "distance not availible"
+        }
+        card.addSubview(distLabel)
+        distLabel.font = Font.thin(12)
+        distLabel.textColor = UIColor.black
+        distLabel.translatesAutoresizingMaskIntoConstraints = false
+        distLabel.topAnchor.constraint(equalTo: phoneLabel.layoutMarginsGuide.bottomAnchor, constant: 20).isActive = true
+        distLabel.leftAnchor.constraint(equalTo: card.layoutMarginsGuide.leftAnchor).isActive = true
+        distLabel.widthAnchor.constraint(equalTo: card.layoutMarginsGuide.widthAnchor, multiplier: 0.5).isActive = true
+        distLabel.adjustsFontSizeToFitWidth = true
+        
+        let libPic = UIImageView()
+        DispatchQueue.global().async {
+            guard let imageData = try? Data(contentsOf: self.library!.imageURL!) else { return }
+            let image = UIImage(data: imageData)
+            DispatchQueue.main.async {
+                libPic.image = image
+            }
+        }
```

Comment:
check if library.image != nil first, image could/should already be saved in the library object, don't need to get it again